backend/controllers/wishlist.controller.js

```diff
@@ -0,0 +1,134 @@
+const Wishlist = require('../models/wishlist.model');
+const Product = require('../models/product.model');
+
+// @desc    Get wishlist by USERID
+// @route   GET /api/wishlists/:id
+// @access  Public
+
+const getWishlistByUserID = async (req, res) => {
+    try {
+
+        const userId = req.user;
+        const findwishlist = await Wishlist.findOne({ user: userId }).populate('productsinWishlist.productId', [
+            'name',
+            'price',
+            'description',
+            'countInStock',
+            'rating',
+            'numReviews',
+            'qty',
+            'category',
+            'image'
+        ]);
+        if (!findwishlist) {
+            return res.status(400).json({ success: false, message: "Wishlist not Found" })
+        }
+        res.status(200).send(findwishlist);
```

Comment:
json object could contain success status as true - to make it consistent with the rest of the responses.

Reply:
So, to make it consistent across all api's 
I have done this
1. If I want to send data, then I have using res.send(data) across all api's, this itself indicates that api call was success
2. If I want to give some message to user based on action, then I am using res.json() with success and message
Thanks